activesupport/lib/active_support/deprecation/reporting.rb

```diff
@@ -36,6 +36,15 @@ def silence
         @silenced = old_silenced
       end
 
+
+      # Outputs a deprecation warning with 'will be removed' message to
```

Comment:
'will be removed' is pretty obtuse. The examples you provide are out of context from people's apps. I.e. I have a hard time seeing how it would make my development easier. And where do I call it?

Let's add an example using a model class:

```ruby
# Outputs a deprecation warning with a timeframe for when the method will be removed.
# Useful to deprecate an unfortunately named method with an optional fallback method for users to call:
#
#   class Post < ActiveRecord::Base
#     def name
#        ActiveSupport::Deprecation.deprecation_warning(__method__, :title)
#        title
#     end
#   end
#
# Logs a deprecation warning: "name is deprecated and will be removed in from Rails 5.0 (use title instead)."
```

Eh, come to think of it, this might not be for users to use in their apps. @matthewd should we just `# :nodoc:` this instead?